From f1671758fc28be21b0470b8bb3ee4d0ebde8c0ae Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Fri, 27 Apr 2018 11:57:55 +0800
Subject: [PATCH] arm: dts: rockchip: Change cpu opp-microvolt form one entry
 to three

Single entry is for target voltage and three entries are for
<target min max> voltages. Change cpu opp-microvolt form one entry to
three entries and set maximum acceptable voltage to a high value so that
regulator device can supply multiple consumers at the same time.

Change-Id: I3a0dc4e161bae33e36b232c36a0a05a3102359ef
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 arch/arm/boot/dts/rk3036.dtsi                 |  10 +-
 arch/arm/boot/dts/rk3066a.dtsi                |   8 +-
 arch/arm/boot/dts/rk312x.dtsi                 |  56 +++----
 arch/arm/boot/dts/rk3229-cpu-opp.dtsi         |  48 +++---
 arch/arm/boot/dts/rk322x.dtsi                 |  30 ++--
 arch/arm/boot/dts/rk3288-android.dtsi         |   2 +-
 arch/arm/boot/dts/rk3288.dtsi                 |  88 +++++------
 arch/arm64/boot/dts/rockchip/px30.dtsi        |  18 +--
 arch/arm64/boot/dts/rockchip/rk3328.dtsi      |  36 ++---
 arch/arm64/boot/dts/rockchip/rk3366.dtsi      |  12 +-
 .../boot/dts/rockchip/rk3399-early-opp.dtsi   |  22 +--
 arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi  | 140 +++++++++---------
 12 files changed, 235 insertions(+), 235 deletions(-)

diff --git a/arch/arm/boot/dts/rk3036.dtsi b/arch/arm/boot/dts/rk3036.dtsi
index e44f8ea77938..b81e135c48fe 100644
--- a/arch/arm/boot/dts/rk3036.dtsi
+++ b/arch/arm/boot/dts/rk3036.dtsi
@@ -95,28 +95,28 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <1000000>;
+			opp-microvolt = <1000000 1000000 1225000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <1000000>;
+			opp-microvolt = <1000000 1000000 1225000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1100000>;
+			opp-microvolt = <1100000 1100000 1225000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1150000>;
+			opp-microvolt = <1150000 1150000 1225000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1225000>;
+			opp-microvolt = <1225000 1225000 1225000>;
 			clock-latency-ns = <40000>;
 		};
 	};
diff --git a/arch/arm/boot/dts/rk3066a.dtsi b/arch/arm/boot/dts/rk3066a.dtsi
index 5df9f1cf1f10..3ac2d2612e4a 100644
--- a/arch/arm/boot/dts/rk3066a.dtsi
+++ b/arch/arm/boot/dts/rk3066a.dtsi
@@ -77,25 +77,25 @@
 
 		opp-312000000 {
 			opp-hz = /bits/ 64 <312000000>;
-			opp-microvolt = <1075000>;
+			opp-microvolt = <1075000 1075000 1125000>;
 			clock-latency-ns = <40000>;
 			status = "disabled";
 		};
 		opp-504000000 {
 			opp-hz = /bits/ 64 <504000000>;
-			opp-microvolt = <1100000>;
+			opp-microvolt = <1100000 1100000 1125000>;
 			clock-latency-ns = <40000>;
 			status = "disabled";
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1125000>;
+			opp-microvolt = <1125000 1125000 1125000>;
 			clock-latency-ns = <40000>;
 			status = "disabled";
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1125000>;
+			opp-microvolt = <1125000 1125000 1125000>;
 			clock-latency-ns = <40000>;
 			status = "disabled";
 		};
diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index 34017fcaa11a..1c5134d97d9e 100644
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -115,59 +115,59 @@
 
 		opp-216000000 {
 			opp-hz = /bits/ 64 <216000000>;
-			opp-microvolt = <950000>;
-			opp-microvolt-L0 = <950000>;
-			opp-microvolt-L1 = <950000>;
-			opp-microvolt-L2 = <950000>;
+			opp-microvolt = <950000 950000 1425000>;
+			opp-microvolt-L0 = <950000 950000 1425000>;
+			opp-microvolt-L1 = <950000 950000 1425000>;
+			opp-microvolt-L2 = <950000 950000 1425000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <950000>;
-			opp-microvolt-L0 = <950000>;
-			opp-microvolt-L1 = <950000>;
-			opp-microvolt-L2 = <950000>;
+			opp-microvolt = <950000 950000 1425000>;
+			opp-microvolt-L0 = <950000 950000 1425000>;
+			opp-microvolt-L1 = <950000 950000 1425000>;
+			opp-microvolt-L2 = <950000 950000 1425000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <1075000>;
-			opp-microvolt-L0 = <1075000>;
-			opp-microvolt-L1 = <1025000>;
-			opp-microvolt-L2 = <975000>;
+			opp-microvolt = <1075000 1075000 1425000>;
+			opp-microvolt-L0 = <1075000 1075000 1425000>;
+			opp-microvolt-L1 = <1025000 1025000 1425000>;
+			opp-microvolt-L2 = <975000 975000 1425000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-696000000 {
 			opp-hz = /bits/ 64 <696000000>;
-			opp-microvolt = <1150000>;
-			opp-microvolt-L0 = <1150000>;
-			opp-microvolt-L1 = <1100000>;
-			opp-microvolt-L2 = <1050000>;
+			opp-microvolt = <1150000 1150000 1425000>;
+			opp-microvolt-L0 = <1150000 1150000 1425000>;
+			opp-microvolt-L1 = <1100000 1100000 1425000>;
+			opp-microvolt-L2 = <1050000 1050000 1425000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1200000>;
-			opp-microvolt-L0 = <1200000>;
-			opp-microvolt-L1 = <1150000>;
-			opp-microvolt-L2 = <1100000>;
+			opp-microvolt = <1200000 1200000 1425000>;
+			opp-microvolt-L0 = <1200000 1200000 1425000>;
+			opp-microvolt-L1 = <1150000 1150000 1425000>;
+			opp-microvolt-L2 = <1100000 1100000 1425000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1325000>;
-			opp-microvolt-L0 = <1325000>;
-			opp-microvolt-L1 = <1275000>;
-			opp-microvolt-L2 = <1225000>;
+			opp-microvolt = <1325000 1325000 1425000>;
+			opp-microvolt-L0 = <1325000 1325000 1425000>;
+			opp-microvolt-L1 = <1275000 1275000 1425000>;
+			opp-microvolt-L2 = <1225000 1225000 1425000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1425000>;
-			opp-microvolt-L0 = <1425000>;
-			opp-microvolt-L1 = <1425000>;
-			opp-microvolt-L2 = <1375000>;
+			opp-microvolt = <1425000 1425000 1425000>;
+			opp-microvolt-L0 = <1425000 1425000 1425000>;
+			opp-microvolt-L1 = <1425000 1425000 1425000>;
+			opp-microvolt-L2 = <1375000 1375000 1425000>;
 			clock-latency-ns = <40000>;
 		};
 	};
diff --git a/arch/arm/boot/dts/rk3229-cpu-opp.dtsi b/arch/arm/boot/dts/rk3229-cpu-opp.dtsi
index e025b05788ce..eac35eb30a4a 100644
--- a/arch/arm/boot/dts/rk3229-cpu-opp.dtsi
+++ b/arch/arm/boot/dts/rk3229-cpu-opp.dtsi
@@ -58,53 +58,53 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <950000>;
-			opp-microvolt-L0 = <950000>;
-			opp-microvolt-L1 = <950000>;
+			opp-microvolt = <950000 950000 1400000>;
+			opp-microvolt-L0 = <950000 950000 1400000>;
+			opp-microvolt-L1 = <950000 950000 1400000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <975000>;
-			opp-microvolt-L0 = <975000>;
-			opp-microvolt-L1 = <975000>;
+			opp-microvolt = <975000 975000 1400000>;
+			opp-microvolt-L0 = <975000 975000 1400000>;
+			opp-microvolt-L1 = <975000 975000 1400000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1000000>;
-			opp-microvolt-L0 = <1000000>;
-			opp-microvolt-L1 = <1000000>;
+			opp-microvolt = <1000000 1000000 1400000>;
+			opp-microvolt-L0 = <1000000 1000000 1400000>;
+			opp-microvolt-L1 = <1000000 1000000 1400000>;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1175000>;
-			opp-microvolt-L0 = <1175000>;
-			opp-microvolt-L1 = <1125000>;
+			opp-microvolt = <1175000 1175000 1400000>;
+			opp-microvolt-L0 = <1175000 1175000 1400000>;
+			opp-microvolt-L1 = <1125000 1125000 1400000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1275000>;
-			opp-microvolt-L0 = <1275000>;
-			opp-microvolt-L1 = <1225000>;
+			opp-microvolt = <1275000 1275000 1400000>;
+			opp-microvolt-L0 = <1275000 1275000 1400000>;
+			opp-microvolt-L1 = <1225000 1225000 1400000>;
 		};
 		opp-1296000000 {
 			opp-hz = /bits/ 64 <1296000000>;
-			opp-microvolt = <1325000>;
-			opp-microvolt-L0 = <1325000>;
-			opp-microvolt-L1 = <1275000>;
+			opp-microvolt = <1325000 1325000 1400000>;
+			opp-microvolt-L0 = <1325000 1325000 1400000>;
+			opp-microvolt-L1 = <1275000 1275000 1400000>;
 		};
 		opp-1392000000 {
 			opp-hz = /bits/ 64 <1392000000>;
-			opp-microvolt = <1375000>;
-			opp-microvolt-L0 = <1375000>;
-			opp-microvolt-L1 = <1325000>;
+			opp-microvolt = <1375000 1375000 1400000>;
+			opp-microvolt-L0 = <1375000 1375000 1400000>;
+			opp-microvolt-L1 = <1325000 1325000 1400000>;
 		};
 		opp-1464000000 {
 			opp-hz = /bits/ 64 <1464000000>;
-			opp-microvolt = <1400000>;
-			opp-microvolt-L0 = <1400000>;
-			opp-microvolt-L1 = <1350000>;
+			opp-microvolt = <1400000 1400000 1400000>;
+			opp-microvolt-L0 = <1400000 1400000 1400000>;
+			opp-microvolt-L1 = <1350000 1350000 1400000>;
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/rk322x.dtsi b/arch/arm/boot/dts/rk322x.dtsi
index c6694b69fa76..7fd83f7fff31 100644
--- a/arch/arm/boot/dts/rk322x.dtsi
+++ b/arch/arm/boot/dts/rk322x.dtsi
@@ -117,35 +117,35 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <950000>;
-			opp-microvolt-L0 = <950000>;
-			opp-microvolt-L1 = <950000>;
+			opp-microvolt = <950000 950000 1275000>;
+			opp-microvolt-L0 = <950000 950000 1275000>;
+			opp-microvolt-L1 = <950000 950000 1275000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <975000>;
-			opp-microvolt-L0 = <975000>;
-			opp-microvolt-L1 = <975000>;
+			opp-microvolt = <975000 975000 1275000>;
+			opp-microvolt-L0 = <975000 975000 1275000>;
+			opp-microvolt-L1 = <975000 975000 1275000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1000000>;
-			opp-microvolt-L0 = <1000000>;
-			opp-microvolt-L1 = <1000000>;
+			opp-microvolt = <1000000 1000000 1275000>;
+			opp-microvolt-L0 = <1000000 1000000 1275000>;
+			opp-microvolt-L1 = <1000000 1000000 1275000>;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1175000>;
-			opp-microvolt-L0 = <1175000>;
-			opp-microvolt-L1 = <1125000>;
+			opp-microvolt = <1175000 1175000 1275000>;
+			opp-microvolt-L0 = <1175000 1175000 1275000>;
+			opp-microvolt-L1 = <1125000 1125000 1275000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1275000>;
-			opp-microvolt-L0 = <1275000>;
-			opp-microvolt-L1 = <1225000>;
+			opp-microvolt = <1275000 1275000 1275000>;
+			opp-microvolt-L0 = <1275000 1275000 1275000>;
+			opp-microvolt-L1 = <1225000 1225000 1275000>;
 		};
 	};
 
diff --git a/arch/arm/boot/dts/rk3288-android.dtsi b/arch/arm/boot/dts/rk3288-android.dtsi
index ddd2a2c5d4ac..1890d5d83398 100644
--- a/arch/arm/boot/dts/rk3288-android.dtsi
+++ b/arch/arm/boot/dts/rk3288-android.dtsi
@@ -403,7 +403,7 @@
 
 	opp-1800000000 {
 		opp-hz = /bits/ 64 <1800000000>;
-		opp-microvolt = <1350000>;
+		opp-microvolt = <1350000 1350000 1350000>;
 		clock-latency-ns = <40000>;
 		status = "disabled";
 	};
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index de94aa450a2b..6e40187a3eac 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -151,91 +151,91 @@
 
 		opp-126000000 {
 			opp-hz = /bits/ 64 <126000000>;
-			opp-microvolt = <900000>;
-			opp-microvolt-L0 = <900000>;
-			opp-microvolt-L1 = <900000>;
-			opp-microvolt-L2 = <900000>;
+			opp-microvolt = <900000 900000 1350000>;
+			opp-microvolt-L0 = <900000 900000 1350000>;
+			opp-microvolt-L1 = <900000 900000 1350000>;
+			opp-microvolt-L2 = <900000 900000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-216000000 {
 			opp-hz = /bits/ 64 <216000000>;
-			opp-microvolt = <900000>;
-			opp-microvolt-L0 = <900000>;
-			opp-microvolt-L1 = <900000>;
-			opp-microvolt-L2 = <900000>;
+			opp-microvolt = <900000 900000 1350000>;
+			opp-microvolt-L0 = <900000 900000 1350000>;
+			opp-microvolt-L1 = <900000 900000 1350000>;
+			opp-microvolt-L2 = <900000 900000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <900000>;
-			opp-microvolt-L0 = <900000>;
-			opp-microvolt-L1 = <900000>;
-			opp-microvolt-L2 = <900000>;
+			opp-microvolt = <900000 900000 1350000>;
+			opp-microvolt-L0 = <900000 900000 1350000>;
+			opp-microvolt-L1 = <900000 900000 1350000>;
+			opp-microvolt-L2 = <900000 900000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <900000>;
-			opp-microvolt-L0 = <900000>;
-			opp-microvolt-L1 = <900000>;
-			opp-microvolt-L2 = <900000>;
+			opp-microvolt = <900000 900000 1350000>;
+			opp-microvolt-L0 = <900000 900000 1350000>;
+			opp-microvolt-L1 = <900000 900000 1350000>;
+			opp-microvolt-L2 = <900000 900000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-696000000 {
 			opp-hz = /bits/ 64 <696000000>;
-			opp-microvolt = <950000>;
-			opp-microvolt-L0 = <950000>;
-			opp-microvolt-L1 = <900000>;
-			opp-microvolt-L2 = <900000>;
+			opp-microvolt = <950000 950000 1350000>;
+			opp-microvolt-L0 = <950000 950000 1350000>;
+			opp-microvolt-L1 = <900000 900000 1350000>;
+			opp-microvolt-L2 = <900000 900000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1050000>;
-			opp-microvolt-L0 = <1050000>;
-			opp-microvolt-L1 = <1000000>;
-			opp-microvolt-L2 = <950000>;
+			opp-microvolt = <1050000 1050000 1350000>;
+			opp-microvolt-L0 = <1050000 1050000 1350000>;
+			opp-microvolt-L1 = <1000000 1000000 1350000>;
+			opp-microvolt-L2 = <950000 950000 1350000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1100000>;
-			opp-microvolt-L0 = <1100000>;
-			opp-microvolt-L1 = <1050000>;
-			opp-microvolt-L2 = <1000000>;
+			opp-microvolt = <1100000 1100000 1350000>;
+			opp-microvolt-L0 = <1100000 1100000 1350000>;
+			opp-microvolt-L1 = <1050000 1050000 1350000>;
+			opp-microvolt-L2 = <1000000 1000000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1150000>;
-			opp-microvolt-L0 = <1150000>;
-			opp-microvolt-L1 = <1100000>;
-			opp-microvolt-L2 = <1050000>;
+			opp-microvolt = <1150000 1150000 1350000>;
+			opp-microvolt-L0 = <1150000 1150000 1350000>;
+			opp-microvolt-L1 = <1100000 1100000 1350000>;
+			opp-microvolt-L2 = <1050000 1050000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1416000000 {
 			opp-hz = /bits/ 64 <1416000000>;
-			opp-microvolt = <1250000>;
-			opp-microvolt-L0 = <1250000>;
-			opp-microvolt-L1 = <1200000>;
-			opp-microvolt-L2 = <1150000>;
+			opp-microvolt = <1250000 1250000 1350000>;
+			opp-microvolt-L0 = <1250000 1250000 1350000>;
+			opp-microvolt-L1 = <1200000 1200000 1350000>;
+			opp-microvolt-L2 = <1150000 1150000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1512000000 {
 			opp-hz = /bits/ 64 <1512000000>;
-			opp-microvolt = <1300000>;
-			opp-microvolt-L0 = <1300000>;
-			opp-microvolt-L1 = <1250000>;
-			opp-microvolt-L2 = <1200000>;
+			opp-microvolt = <1300000 1300000 1350000>;
+			opp-microvolt-L0 = <1300000 1300000 1350000>;
+			opp-microvolt-L1 = <1250000 1250000 1350000>;
+			opp-microvolt-L2 = <1200000 1200000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1608000000 {
 			opp-hz = /bits/ 64 <1608000000>;
-			opp-microvolt = <1350000>;
-			opp-microvolt-L0 = <1350000>;
-			opp-microvolt-L1 = <1300000>;
-			opp-microvolt-L2 = <1250000>;
+			opp-microvolt = <1350000 1350000 1350000>;
+			opp-microvolt-L0 = <1350000 1350000 1350000>;
+			opp-microvolt-L1 = <1300000 1300000 1350000>;
+			opp-microvolt-L2 = <1250000 1250000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/rockchip/px30.dtsi b/arch/arm64/boot/dts/rockchip/px30.dtsi
index 3c48d4bb528c..1b99e2668f71 100644
--- a/arch/arm64/boot/dts/rockchip/px30.dtsi
+++ b/arch/arm64/boot/dts/rockchip/px30.dtsi
@@ -116,48 +116,48 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <950000>;
+			opp-microvolt = <950000 950000 1350000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <950000>;
+			opp-microvolt = <950000 950000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1000000>;
+			opp-microvolt = <1000000 1000000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1125000>;
+			opp-microvolt = <1125000 1125000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1275000>;
+			opp-microvolt = <1275000 1275000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1248000000 {
 			opp-hz = /bits/ 64 <1248000000>;
-			opp-microvolt = <1300000>;
+			opp-microvolt = <1300000 1300000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1296000000 {
 			opp-hz = /bits/ 64 <1296000000>;
-			opp-microvolt = <1350000>;
+			opp-microvolt = <1350000 1350000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1416000000 {
 			opp-hz = /bits/ 64 <1416000000>;
-			opp-microvolt = <1350000>;
+			opp-microvolt = <1350000 1350000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1512000000 {
 			opp-hz = /bits/ 64 <1512000000>;
-			opp-microvolt = <1350000>;
+			opp-microvolt = <1350000 1350000 1350000>;
 			clock-latency-ns = <40000>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/rockchip/rk3328.dtsi b/arch/arm64/boot/dts/rockchip/rk3328.dtsi
index f405673580be..4dc0915790d8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3328.dtsi
@@ -120,45 +120,45 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <975000>;
-			opp-microvolt-L0 = <975000>;
-			opp-microvolt-L1 = <950000>;
+			opp-microvolt = <975000 975000 1325000>;
+			opp-microvolt-L0 = <975000 975000 1325000>;
+			opp-microvolt-L1 = <950000 950000 1325000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <975000>;
-			opp-microvolt-L0 = <975000>;
-			opp-microvolt-L1 = <950000>;
+			opp-microvolt = <975000 975000 1325000>;
+			opp-microvolt-L0 = <975000 975000 1325000>;
+			opp-microvolt-L1 = <950000 950000 1325000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1025000>;
-			opp-microvolt-L0 = <1025000>;
-			opp-microvolt-L1 = <1000000>;
+			opp-microvolt = <1025000 1025000 1325000>;
+			opp-microvolt-L0 = <1025000 1025000 1325000>;
+			opp-microvolt-L1 = <1000000 1000000 1325000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1125000>;
-			opp-microvolt-L0 = <1125000>;
-			opp-microvolt-L1 = <1100000>;
+			opp-microvolt = <1125000 1125000 1325000>;
+			opp-microvolt-L0 = <1125000 1125000 1325000>;
+			opp-microvolt-L1 = <1100000 1100000 1325000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1250000>;
-			opp-microvolt-L0 = <1250000>;
-			opp-microvolt-L1 = <1225000>;
+			opp-microvolt = <1250000 1250000 1325000>;
+			opp-microvolt-L0 = <1250000 1250000 1325000>;
+			opp-microvolt-L1 = <1225000 1225000 1325000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1296000000 {
 			opp-hz = /bits/ 64 <1296000000>;
-			opp-microvolt = <1325000>;
-			opp-microvolt-L0 = <1325000>;
-			opp-microvolt-L1 = <1300000>;
+			opp-microvolt = <1325000 1325000 1325000>;
+			opp-microvolt-L0 = <1325000 1325000 1325000>;
+			opp-microvolt-L1 = <1300000 1300000 1325000>;
 			clock-latency-ns = <40000>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/rockchip/rk3366.dtsi b/arch/arm64/boot/dts/rockchip/rk3366.dtsi
index ca401d68c365..927daa725235 100644
--- a/arch/arm64/boot/dts/rockchip/rk3366.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3366.dtsi
@@ -147,29 +147,29 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <950000>;
+			opp-microvolt = <950000 950000 1250000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <950000>;
+			opp-microvolt = <950000 950000 1250000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1000000>;
+			opp-microvolt = <1000000 1000000 1250000>;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1075000>;
+			opp-microvolt = <1075000 1075000 1250000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1175000>;
+			opp-microvolt = <1175000 1175000 1250000>;
 		};
 		opp-1296000000 {
 			opp-hz = /bits/ 64 <1296000000>;
-			opp-microvolt = <1250000>;
+			opp-microvolt = <1250000 1250000 1250000>;
 		};
 	};
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-early-opp.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-early-opp.dtsi
index ae1007705f36..2a675ec7c342 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-early-opp.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-early-opp.dtsi
@@ -56,28 +56,28 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <900000>;
+			opp-microvolt = <900000 900000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <900000>;
+			opp-microvolt = <900000 900000 1200000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <900000>;
+			opp-microvolt = <900000 900000 1200000>;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <900000>;
+			opp-microvolt = <900000 900000 1200000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <950000>;
+			opp-microvolt = <950000 950000 1200000>;
 		};
 		opp-1416000000 {
 			opp-hz = /bits/ 64 <1416000000>;
-			opp-microvolt = <1050000>;
+			opp-microvolt = <1050000 1050000 1200000>;
 		};
 	};
 
@@ -87,24 +87,24 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <900000>;
+			opp-microvolt = <900000 900000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <900000>;
+			opp-microvolt = <900000 900000 1200000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <900000>;
+			opp-microvolt = <900000 900000 1200000>;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <950000>;
+			opp-microvolt = <950000 950000 1200000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <100000>;
+			opp-microvolt = <1000000 1000000 1200000>;
 		};
 	};
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
index e8599ed532c0..f443a9a21f98 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
@@ -68,57 +68,57 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <800000>;
-			opp-microvolt-L0 = <800000>;
-			opp-microvolt-L1 = <800000>;
-			opp-microvolt-L2 = <800000>;
-			opp-microvolt-L3 = <800000>;
+			opp-microvolt = <800000 800000 1200000>;
+			opp-microvolt-L0 = <800000 800000 1200000>;
+			opp-microvolt-L1 = <800000 800000 1200000>;
+			opp-microvolt-L2 = <800000 800000 1200000>;
+			opp-microvolt-L3 = <800000 800000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <800000>;
-			opp-microvolt-L0 = <800000>;
-			opp-microvolt-L1 = <800000>;
-			opp-microvolt-L2 = <800000>;
-			opp-microvolt-L3 = <800000>;
+			opp-microvolt = <800000 800000 1200000>;
+			opp-microvolt-L0 = <800000 800000 1200000>;
+			opp-microvolt-L1 = <800000 800000 1200000>;
+			opp-microvolt-L2 = <800000 800000 1200000>;
+			opp-microvolt-L3 = <800000 800000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <850000>;
-			opp-microvolt-L0 = <850000>;
-			opp-microvolt-L1 = <825000>;
-			opp-microvolt-L2 = <800000>;
-			opp-microvolt-L3 = <800000>;
+			opp-microvolt = <850000 850000 1200000>;
+			opp-microvolt-L0 = <850000 850000 1200000>;
+			opp-microvolt-L1 = <825000 825000 1200000>;
+			opp-microvolt-L2 = <800000 800000 1200000>;
+			opp-microvolt-L3 = <800000 800000 1200000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <925000>;
-			opp-microvolt-L0 = <925000>;
-			opp-microvolt-L1 = <900000>;
-			opp-microvolt-L2 = <875000>;
-			opp-microvolt-L3 = <850000>;
+			opp-microvolt = <925000 925000 1200000>;
+			opp-microvolt-L0 = <925000 925000 1200000>;
+			opp-microvolt-L1 = <900000 900000 1200000>;
+			opp-microvolt-L2 = <875000 875000 1200000>;
+			opp-microvolt-L3 = <850000 850000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1000000>;
-			opp-microvolt-L0 = <1000000>;
-			opp-microvolt-L1 = <975000>;
-			opp-microvolt-L2 = <950000>;
-			opp-microvolt-L3 = <925000>;
+			opp-microvolt = <1000000 1000000 1200000>;
+			opp-microvolt-L0 = <1000000 1000000 1200000>;
+			opp-microvolt-L1 = <975000 975000 1200000>;
+			opp-microvolt-L2 = <950000 950000 1200000>;
+			opp-microvolt-L3 = <925000 925000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1416000000 {
 			opp-hz = /bits/ 64 <1416000000>;
-			opp-microvolt = <1125000>;
-			opp-microvolt-L0 = <1125000>;
-			opp-microvolt-L1 = <1100000>;
-			opp-microvolt-L2 = <1075000>;
-			opp-microvolt-L3 = <1050000>;
+			opp-microvolt = <1125000 1125000 1200000>;
+			opp-microvolt-L0 = <1125000 1125000 1200000>;
+			opp-microvolt-L1 = <1100000 1100000 1200000>;
+			opp-microvolt-L2 = <1075000 1075000 1200000>;
+			opp-microvolt-L3 = <1050000 1050000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 	};
@@ -148,75 +148,75 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <800000>;
-			opp-microvolt-L0 = <800000>;
-			opp-microvolt-L1 = <800000>;
-			opp-microvolt-L2 = <800000>;
-			opp-microvolt-L3 = <800000>;
+			opp-microvolt = <800000 800000 1200000>;
+			opp-microvolt-L0 = <800000 800000 1200000>;
+			opp-microvolt-L1 = <800000 800000 1200000>;
+			opp-microvolt-L2 = <800000 800000 1200000>;
+			opp-microvolt-L3 = <800000 800000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <800000>;
-			opp-microvolt-L0 = <800000>;
-			opp-microvolt-L1 = <800000>;
-			opp-microvolt-L2 = <800000>;
-			opp-microvolt-L3 = <800000>;
+			opp-microvolt = <800000 800000 1200000>;
+			opp-microvolt-L0 = <800000 800000 1200000>;
+			opp-microvolt-L1 = <800000 800000 1200000>;
+			opp-microvolt-L2 = <800000 800000 1200000>;
+			opp-microvolt-L3 = <800000 800000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <825000>;
-			opp-microvolt-L0 = <825000>;
-			opp-microvolt-L1 = <825000>;
-			opp-microvolt-L2 = <800000>;
-			opp-microvolt-L3 = <800000>;
+			opp-microvolt = <825000 825000 1200000>;
+			opp-microvolt-L0 = <825000 825000 1200000>;
+			opp-microvolt-L1 = <825000 825000 1200000>;
+			opp-microvolt-L2 = <800000 800000 1200000>;
+			opp-microvolt-L3 = <800000 800000 1200000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <875000>;
-			opp-microvolt-L0 = <875000>;
-			opp-microvolt-L1 = <850000>;
-			opp-microvolt-L2 = <850000>;
-			opp-microvolt-L3 = <850000>;
+			opp-microvolt = <875000 875000 1200000>;
+			opp-microvolt-L0 = <875000 875000 1200000>;
+			opp-microvolt-L1 = <850000 850000 1200000>;
+			opp-microvolt-L2 = <850000 850000 1200000>;
+			opp-microvolt-L3 = <850000 850000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <950000>;
-			opp-microvolt-L0 = <950000>;
-			opp-microvolt-L1 = <925000>;
-			opp-microvolt-L2 = <900000>;
-			opp-microvolt-L3 = <875000>;
+			opp-microvolt = <950000 950000 1200000>;
+			opp-microvolt-L0 = <950000 950000 1200000>;
+			opp-microvolt-L1 = <925000 925000 1200000>;
+			opp-microvolt-L2 = <900000 900000 1200000>;
+			opp-microvolt-L3 = <875000 875000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1416000000 {
 			opp-hz = /bits/ 64 <1416000000>;
-			opp-microvolt = <1025000>;
-			opp-microvolt-L0 = <1025000>;
-			opp-microvolt-L1 = <1000000>;
-			opp-microvolt-L2 = <1000000>;
-			opp-microvolt-L3 = <975000>;
+			opp-microvolt = <1025000 1025000 1200000>;
+			opp-microvolt-L0 = <1025000 1025000 1200000>;
+			opp-microvolt-L1 = <1000000 1000000 1200000>;
+			opp-microvolt-L2 = <1000000 1000000 1200000>;
+			opp-microvolt-L3 = <975000 975000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1608000000 {
 			opp-hz = /bits/ 64 <1608000000>;
-			opp-microvolt = <1100000>;
-			opp-microvolt-L0 = <1100000>;
-			opp-microvolt-L1 = <1075000>;
-			opp-microvolt-L2 = <1050000>;
-			opp-microvolt-L3 = <1025000>;
+			opp-microvolt = <1100000 1100000 1200000>;
+			opp-microvolt-L0 = <1100000 1100000 1200000>;
+			opp-microvolt-L1 = <1075000 1075000 1200000>;
+			opp-microvolt-L2 = <1050000 1050000 1200000>;
+			opp-microvolt-L3 = <1025000 1025000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1800000000 {
 			opp-hz = /bits/ 64 <1800000000>;
-			opp-microvolt = <1200000>;
-			opp-microvolt-L0 = <1200000>;
-			opp-microvolt-L1 = <1175000>;
-			opp-microvolt-L2 = <1150000>;
-			opp-microvolt-L3 = <1125000>;
+			opp-microvolt = <1200000 1200000 1200000>;
+			opp-microvolt-L0 = <1200000 1200000 1200000>;
+			opp-microvolt-L1 = <1175000 1175000 1200000>;
+			opp-microvolt-L2 = <1150000 1150000 1200000>;
+			opp-microvolt-L3 = <1125000 1125000 1200000>;
 			clock-latency-ns = <40000>;
 		};
 	};
-- 
2.35.3

